src/DotNetLightning.Core/LN/Channel.fs

```diff
@@ -42,11 +42,11 @@ type Channel = {
             }
         static member CreateCurried = curry9 (Channel.Create)
 
-type ChannelError =
-    | Ignore of string
-    | Close of string
+type CloseChannelException(msg: string, stackTrace: string) =
+    inherit System.Exception(msg)
+    override this.StackTrace = stackTrace
```

Comment:
I've never seen this weird pattern used in .NET; if you raise an exception and capture it with a `try` block you will achieve getting line numbers and stacktraces automatically

Reply:
the exception is never raised, it is simply constructed. That's why the stack trace will be empty.

Reply:
right, so why not raise it instead?

Reply:
That is a larger refactor. If exceptions were used, there would be no need for the RBadTree and such, I guess. I don't know the motivation for avoiding exceptions, maybe @joemphilips can chime in.

Reply:
Good point, let's hear Joe.

Reply:
Well, I just wanted to make it explicit that the function is not safe from the callers point of view.
(I usually don't prefer throwing exception because consumer of the api will often forgets handling the exception.) So it is just typical motivation for using Result.

But you are right that debugging is pain when we use `RResult` naively. 

I think this fix is good enough for now. But ideally we should avoid exposing RResult to the public api all together and prepare exception type for each domain `ChannelError` `PeerError`
And each of them will hold stacktrace info (or some sort.)

Reply:
> prepare exception type for each domain `ChannelError` `PeerError`

To make myself clear, this means model error type by DU (instead of exception) and hold stacktrace-like info in its type signature.

Reply:
all right. I think exceptions don't mix well with functional programming because they are not reflected in the type signatures, as you mention. With a sufficiently granular DU maybe we can forget about source locations because there will only be a single hit when you grep for it. Anyway,  how do I proceed with this PR?

Reply:
wouldn't sufficient logging in case of errors be enough? so long as you use different logging messages for each location

Reply:
just a sec in will review more closely in an hour

Reply:
Yes, lets keep using `ChannelError` instead of channel exception.
Using exception in this way does not seem to help.

You can change the signature as you want, adding file number field to ChannelError seems not bad for improving debugging experience.

and as @knocte suggested, logging is a good option too.

So you can do either of
1. use ChannelError and change its signature to `lineNumber: string * msg : string` for each case.
2. do nothing (close PR)
3. try refactoring out RResult and have more granular cases for ChannelError. (This is a big change and I might do it by myself later but I'd appreciate if you try it.)